plugins/BEdita/Core/src/Model/Behavior/SearchableBehavior.php

```diff
@@ -29,84 +32,93 @@
+
+    /**
+     * Update search adapters index when a resource is saved.
      *
-     * If fields or column types are specified - do *not* merge them with existing config,
-     * overwrite the fields to search on.
+     * @param \Cake\Event\EventInterface $event The event
+     * @param \Cake\Datasource\EntityInterface $entity The resource entity
+     * @return void
      */
-    public function initialize(array $config): void
+    public function afterSave(EventInterface $event, EntityInterface $entity): void
     {
-        foreach (['columnTypes', 'fields'] as $key) {
-            if (isset($config[$key])) {
-                $this->setConfig($key, $config[$key], false);
-            }
+        foreach (array_keys((array)Configure::read('Search.adapters')) as $name) {
+            $this->getAdapter($name)->indexResource($entity, 'edit');
         }
     }
 
     /**
-     * Get all fields whose column type is amongst those allowed in `columnTypes` configuration key.
+     * Update search adapters index when a resource is deleted.
      *
-     * @param \Cake\ORM\Table $table Table object.
-     * @return string[]
+     * @param \Cake\Event\EventInterface $event The event
+     * @param \Cake\Datasource\EntityInterface $entity The resource entity
+     * @return void
      */
-    protected function getAllFields(Table $table)
+    public function afterDelete(EventInterface $event, EntityInterface $entity): void
     {
-        $columnTypes = $this->getConfig('columnTypes');
-        $fields = array_filter( // Filter fields that are of a searchable type.
-            $table->getSchema()->columns(),
-            function ($column) use ($columnTypes, $table) {
-                return in_array($table->getSchema()->getColumnType($column), $columnTypes);
-            }
-        );
+        foreach (array_keys((array)Configure::read('Search.adapters')) as $name) {
+            $this->getAdapter($name)->indexResource($entity, 'delete');
+        }
+    }
 
-        if ($table instanceof InheritanceTable && $table->inheritedTable() !== null) {
-            // If table inherits from another table, merge parent table's fields.
-            $fields = array_merge($fields, $this->getAllFields($table->inheritedTable()));
+    /**
+     * Get search adapters registry.
+     *
+     * @return \BEdita\Core\Search\SearchRegistry
+     */
+    protected function getSearchRegistry(): SearchRegistry
+    {
+        if (!isset($this->searchRegistry)) {
+            $this->searchRegistry = new SearchRegistry();
         }
 
-        return $fields;
+        return $this->searchRegistry;
     }
 
     /**
-     * Get searchable fields and their priorities.
+     * Get an adapter by name.
      *
-     * @return array Array where keys are columns, and values are priorities.
+     * @param string $name The adapter name
+     * @return \BEdita\Core\Search\BaseAdapter
      */
-    public function getFields()
+    protected function getAdapter($name): BaseAdapter
```

Comment:
```suggestion
    protected function getAdapter(string $name): BaseAdapter
```

I also wonder if we should allow `null` and fallback to the "default" adapter, for searching purposes:

```php
$myAdapter = $this->getAdapter('foo');
$defaultAdapter = $this->getAdapter(); // or $this->getAdapter(null);
```